### DISCUSSION TITLE
What's the proper way to deal with secondary programs?

### DISCUSSION THREAD

**Discussion by github-account1111** [Klocman/Bulk-Crap-Uninstaller#604] [Q&A] [answered]
Some programs have "secondary" entries in BCU:

![image](https://github.com/Klocman/Bulk-Crap-Uninstaller/assets/5728319/797b69e8-7e4c-40b6-9370-472bb5daaa1d)
How should I go about the order of uninstallation?

For Beyond Compare, it's obvious what the main program is. Should I uninstall that or the secondary "shell extension" first?

For WinMerge it's not even clear which is the main program.

The Installed apps list has both entries for Beyond Compare, but only one entry for WinMerge, whereas Programs & Features has one entry for *both* which adds to the confusion.

**Comment by Klocman** [comment, marked as answer]
Generally it's fine to select everything and let BCU figure it out. If you pick only one of the items BCU will ask you to also add the other one anyways.
You can uncheck everything except for registry entries in detection settings to see roughly the same list as in Programs & Features.
